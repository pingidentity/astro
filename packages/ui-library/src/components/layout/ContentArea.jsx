import React from "react";
import classnames from "classnames";

/**
 * @class ContentArea
 * @desc The main content area below the HeaderBar and next to the LeftNavBar
 *
 * @param {string} [data-id]
 *      To define the base "data-id" value for the top-level HTML container.
 * @param {string} [className]
 *     Class name(s) to add to the top-level container/div.
 */

const ContentArea = ({
    children,
    className,
    "data-id": dataId,
}) => (<div data-id={dataId} className={classnames("content-area", className)}>{children}</div>);

export default ContentArea;