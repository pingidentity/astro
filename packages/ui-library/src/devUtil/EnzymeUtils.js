import { mount } from "enzyme";


export const snapshotDataIds = node => {
    const elements = node.find("[data-id]").hostNodes(); // Find only actual HTML elements

    const ids = elements.map(element => element.prop("data-id")).filter(item => item !== "");
    expect(ids).toMatchSnapshot();
};

export const mountSnapshotDataIds = node => snapshotDataIds(mount(node));
