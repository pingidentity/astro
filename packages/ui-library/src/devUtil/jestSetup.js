
// This is needed for testing the modal and DetailsTooltip component.
global.getSelection = jest.fn();
global.getSelection.mockReturnValue({
    toString: () => "",
});
global.document.createRange = () => ({
    setStart: () => {},
    setEnd: () => {},
    commonAncestorContainer: {
        nodeName: "BODY",
        ownerDocument: document,
    },
});
