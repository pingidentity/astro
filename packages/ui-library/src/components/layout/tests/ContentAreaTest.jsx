window.__DEV__ = true;

import React from "react";
import { mountSnapshotDataIds } from "../../../devUtil/EnzymeUtils";
import ContentArea from "../ContentArea";

describe("ContentArea", function () {

    it("data-id's don't change", () => {
        mountSnapshotDataIds(
            <ContentArea
                onCancel={jest.fn()}
            />
        );
    });

});