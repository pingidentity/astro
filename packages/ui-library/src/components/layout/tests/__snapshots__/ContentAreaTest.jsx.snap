// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentArea data-id's don't change 1`] = `Array []`;
